### PR TITLE
MINOR: Replacing for with foreach loop in common module

### PR DESCRIPTION
**Pull request by kashyap-prabhat**

### CODE REVIEW

File: clients/src/main/java/org/apache/kafka/common/metrics/Sensor.java [hunk 1]
```diff
@@ -117,8 +117,7 @@ public boolean shouldRecord(final int configId) {
     private void checkForest(Set<Sensor> sensors) {
         if (!sensors.add(this))
             throw new IllegalArgumentException("Circular dependency in sensors: " + name() + " is its own parent.");
-        for (int i = 0; i < parents.length; i++)
-            parents[i].checkForest(sensors);
+        for (Sensor parent : parents) parent.checkForest(sensors);
```
Comment (ijuma) [inline clients/src/main/java/org/apache/kafka/common/metrics/Sensor.java]: The convention (even though not enforced by checkstyle) is for `parent.checkForest(sensors);` to go in the next line for the Java code.

**Review by ijuma — COMMENTED** [review]
Thanks for the PR. The usage of the foreach loop looks good. However, there are a number of changes where the newline is removed after the foreach or if/else and we don't do that in Kafka's Java code (although we do in Scala's code).

**Comment by kashyap-prabhat** [comment]
@ijuma Thanks for pointing it out. Please check I have fixed it.

**Review by ijuma — APPROVED** [review]
Thanks for the PR, LGTM. I fixed formatting in a few places before merging to trunk.

**Pull request closed without merging**